agent/hcp/client/metrics_client.go

```diff
@@ -139,6 +139,10 @@ func (o *otlpClient) ExportMetrics(ctx context.Context, protoMetrics *metricpb.R
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
```

Comment:
I hit a case where I would get a `404: not found`because the endpoint wasn't hooked up. This made the code continue and try to deserialize an object that wasn't there. So Added an early return if the status isn't `200: OK`